### DISCUSSION TITLE
Git submodule -> Show list of changes

### DISCUSSION THREAD

**Discussion by psmit** [renovatebot/renovate#30231] [Suggest an Idea]
### Tell us more.

It would be great if, when the number of commits is reasonable (and there is a simple progression), the list of new commits between the old and new ref are shown in the update merge request for a git submodule.

**Comment by HonkingGoose** [comment]
We had issue #7776, which got converted into discussion #24903.

Summary of the ideas in that discussion:

- Parse around `git diff --submodule=diff`, and show output.
- Add a `diff` compare link to the Renovate PR for Git submodule update. I guess the maintainer meant a link to the GitHub diff page.
- Show `git submodule summary` output.
- Show `git submodule summary --summary-limit` output. The flag limits the summary to `x` number of commits.

`@rarkins` said they stopped listing commits in Renovate's PR body text, because the list added noise:

> This is not really a changelog by the usual definition. Sounds like you want a list of commits, which could also apply to other types of updates too. We stopped including them in repos at one point because it can introduce a huge amount of noise in some cases

I don't know which of the ideas are good, or if we even want to build the feature. So please wait for a response from the maintainer. :wink:

**Reply by psmit** [reply]
Thank you so much for your extensive reply. Unfortunately, I had not found the earlier discussion. I'll look if there is anything I can contribute there.

**Comment by jamietanna** [comment]
The `changelogUrl` is now auto-populated when using [`helpers:githubDigestChangelogs`](https://docs.renovatebot.com/presets-helpers/#helpersgithubdigestchangelogs) (included in `config:recommended`)

_However_ Git submodules don't currently set the `sourceUrl`, so I've raised https://github.com/renovatebot/renovate/pull/40472 for this

Until then, you can do i.e.

```json
{
	"$schema": "https://docs.renovatebot.com/renovate-schema.json",
	"extends": [
		"helpers:githubDigestChangelogs"
	],
	"git-submodules": {
		"enabled": true
	},
	"packageRules": [
		{
			"matchPackageNames": [
				"https://github.com/renovatebot/renovate.git"
			],
			"matchManagers": [
				"git-submodules"
			],
			"sourceUrl": "https://github.com/renovatebot/renovate.git"
		}
	]
}
```
